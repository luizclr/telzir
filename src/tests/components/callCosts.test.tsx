import React from "react";
import CallCosts from "../../components/callCosts";
import { shallow } from "enzyme";

describe("<CallCosts />", () => {
  it("should render", () => {
    const wrapper = shallow(<CallCosts />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should render with VALID information", () => {
    const props = { valid: true, withPlan: 0, withoutPlan: 0 };
    const wrapper = shallow(<CallCosts callCosts={props} />);

    expect(wrapper.find("#withPlan").text()).toEqual("$ 0.00");
    expect(wrapper.find("#withoutPlan").text()).toEqual("$ 0.00");
  });

  it("should render with INVALID information", () => {
    const props = { valid: false, withPlan: 0, withoutPlan: 0 };
    const wrapper = shallow(<CallCosts callCosts={props} />);

    expect(wrapper.find("#withPlan").text()).toEqual("-");
    expect(wrapper.find("#withoutPlan").text()).toEqual("-");
  });
});
