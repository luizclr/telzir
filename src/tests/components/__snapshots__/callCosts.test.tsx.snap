// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CallCosts /> should render 1`] = `
<div
  className="call-costs"
>
  <div
    className="call-costs__row"
  >
    <p>
      With plan:
    </p>
    <span
      id="withPlan"
    >
      -
    </span>
  </div>
  <div
    className="call-costs__row"
  >
    <p>
      Without plan:
    </p>
    <span
      id="withoutPlan"
    >
      -
    </span>
  </div>
</div>
`;
